build.gradle.kts

```diff
@@ -46,6 +66,24 @@ subprojects {
     resolutionStrategy.force(groovy.util.Eval.x(this@subprojects, "x.dep.errorProneCore"))
   }
 
+  // TODO: Make this lazy
+  this@subprojects.afterEvaluate {
+    val jarTask = this@subprojects.tasks.findByName("jar") as? org.gradle.jvm.tasks.Jar
```

Comment:
You could remove afterEvaluate by doing something like this but it will not be lazy since it will configure the jar task, regardless. 

```
this@subprojects.tasks.matching { it.name == "jar" }.all { jarTask ->
}

```

Reply:
I actually tried something similar (with tasks.withType instead of tasks.matching) and it did not got well :-/. It failed on:

```
* What went wrong:
A problem occurred configuring project ':apollo-android-support'.
> Could not create task ':apollo-android-support:androidJavadocsJar'.
   > DefaultTaskContainer#register(String, Class, Action) on task set cannot be executed in the current context.
```

I'm not sure what's happening exactly. Maybe creating a task from another task configuration isn't allowed. That wouldn't surprise me too much.

Reply:
Oh yeah, I hit this before. It is not allowed. As far as I learned the solution is to create dummy tasks that just depends on the real task based on `jar` being available. 

So you always create the task but only run if conditions are met.

Reply:
I must be missing something, even calling `dependsOn` is not allowed it seems: https://github.com/martinbonnin/apollo-android/blob/94ae0e40777372218fe3e4fb5f2ccfcc0578ee33/build.gradle.kts#L87

Reply:
Ok, I needed to do everything from `all` and not `configureEach`: https://github.com/martinbonnin/apollo-android/commit/9a143e118272c23e514a7bcedcd03f0706064f97

Reply:
Ok so using `all` doesn't work as it's eager so it's actually worse than everything else as it forces the configuration of *all* the tasks...

Reply:
`findByName` would also force the configuration of all the tasks, no?

Reply:
> findByName would also force the configuration of all the tasks, no?

Indeed. The problem is that it forces configuring the actual Task which assumes a `jar` task is present. Oh well, I guess I can skip configuration in that case...

Reply:
But... since `all` forces the configuration of the `actualJapicmp`task, it gets configured before the `jar` task is added by the gradle-mvn-push script and we're back to `project.afterEvaluate {}`...